### PR TITLE
Demo ui

### PR DESCRIPTION
**Pull request by abeglova**
#### Pre-Flight checklist

- [ ] Screenshots and design review for any changes that affect layout or styling
Screen Shot 2023-04-06 at 8 23 11 AM
Screen Shot 2023-04-06 at 8 23 28 AM
Screen Shot 2023-04-06 at 8 23 50 AM
Screen Shot 2023-04-06 at 8 24 10 AM

- [ ] Migrations
  - [ ] Migration is backwards-compatible with current production code
- [ ] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested
- [ ] Settings
  - [ ] New settings are documented and present in `app.json`
  - [ ] New settings have reasonable development defaults, if applicable

#### What are the relevant tickets?
https://github.com/mitodl/open-discussions/issues/3909

#### What's this PR do?
This pr creates a simplified version of the search for the innovation mart.

The demo search cards are deliberately not clickable

#### How should this be manually tested?
Go to http://localhost:8063/infinite/demo
Click around

### CODE REVIEW

File: frontends/infinite-corridor/src/pages/DemoPage.test.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    expect(makeRequest.mock.calls[0][0]).toEqual("post")
+    expect(makeRequest.mock.calls[0][1]).toEqual("search/")
+    expect(makeRequest.mock.calls[0][2]).toMatchObject(
+      buildSearchQuery({
+        text:         "New Search Text",
+        from:         0,
+        activeFacets: { type: ["course"] },
+        size:         5
+      })
+    )
```
Comment (ChristopherChudzicki) [inline frontends/infinite-corridor/src/pages/DemoPage.test.tsx:31]: You can also spy on `axios.post` now if you prefer:
```ts
import axios from "../../lib/axios"
// ... then later:
expect(axios.post).toHaveBeenNthCalledWith(0, "/search", body)
```
(*Now you can; last summer this did not work.*)

File: frontends/infinite-corridor/src/pages/DemoPage.tsx [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        from,
+        activeFacets: podcastFilter,
+        size:         pageSize
+      })
+
+      setRequestInFlight(false)
+
+      if (
+        !newCourseResults ||
+        newCourseResults["apiFailed"] ||
```
Comment (ChristopherChudzicki) [inline frontends/infinite-corridor/src/pages/DemoPage.tsx:103]: Can the results include "apiFailed" ?

**Review by ChristopherChudzicki — APPROVED** [review]
👍 Looks good to me.

I left some suggestions. Up to you if you want to do them at all, here, or separate PR.

I do think the `Promise.all` thing is definitely worth doing... It's easy and the page is noticeably more performant with that.

File: frontends/infinite-corridor/src/pages/DemoPage.tsx (hunk 2 above)
Comment (abeglova) [reply frontends/infinite-corridor/src/pages/DemoPage.tsx:103]: I'm not sure where this is coming from. We check for it in ocw-hugo-themes as well https://github.com/mitodl/ocw-hugo-themes/pull/662/files#diff-b681b498395009bcd9a1b7c4c214db07e6997fd551aec850c8a026447a7e70f2R66

**Review by ChristopherChudzicki — APPROVED** [review]
LGTM

**Review by ChristopherChudzicki — APPROVED** [review]

**Pull request merged**